htdocs/variants/class/ProductCombination.class.php

```diff
@@ -527,10 +527,10 @@ public function updateProperties(Product $parent, User $user)
 		$child->price_autogen = $parent->price_autogen;
 		$child->weight = $parent->weight;
 		// Only when Parent Status are updated
-		if (!empty($parent->oldcopy) && ($parent->status != $parent->oldcopy->status)) {
+		if (($parent->oldcopy !== null) && ($parent->status != $parent->oldcopy->status)) {
```

Comment:
Can't see why we can't use 
!empty($parent->oldcopy)
Whch value do you have for ->oldcopy that return a different value with !== null ?

Reply:
Indeed it's a recommendation from PHPStan not to use empty() here.

If you try this simple code on https://phpstan.org/try it will throw a PHPStan error:
```
<?php
class Product { public Product $oldcopy; }
$product = new Product();
$product->oldcopy = $product;
echo empty($product->oldcopy);
```
&rarr; Line: 5. Error: "Property Product::$oldcopy (Product) in empty() is not falsy."    [empty.property](https://phpstan.org/error-identifiers#gbi-empty.property)

And if you also tick the box "Strict rules", PHPStan will throw a second error:
&rarr; Line: 5. Error: Construct empty() is not allowed. Use more strict comparison.     [empty.notAllowed](https://phpstan.org/error-identifiers#gbi-empty.notallowed)


PHPStan probably recommends that to minimize the side effects with `empty()`.
I've check on the Internet and found this one described [here](https://localheinz.com/articles/2023/05/10/avoiding-empty-in-php/#content-inaccessible-instance-property) where the magic method `__isset()` is defined for the object ; and the property is either protected or private.

For example the code below returns `true`:
```
<?php
class Product { 
	protected int $num = 1;
	public function __isset(string $name):bool { return true; }
}
$product = new Product();
var_dump(empty($product->num));
```

There might be other side effects.

Bottom line is: I think PHPStan's recommendation is to be cristal clear on what we're testing.
But PHPStan might be as well overreacting.

@mdeweerd : any idea?